submissions/Temu4/tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,74 @@
+// ======== OBJECTS DEFINITIONS ========
+class Creature {
+  constructor(species, gender, name, age, hobby, legs) {
+    this.species = species;
+    this.gender = gender;
+    this.name = name;
+    this.age = age;
+    this.legs = legs;
+    this.hobby = hobby;
+  }
+}
+
+class Human extends Creature {
+  constructor(gender, name, age, hobby, legs = 2, hands) {
+    super('human', gender, name, age, hobby, legs);
+    this.hands = 2;
+  }
+}
+
+class Man extends Human {
+  constructor(name, age, hobby) {
+    super('male', name, age, hobby);
+  }
+  greeting() {
+    return `\tHello! My name is <strong>${this.name}</strong>. I'm ${this.age}. Little about me:
+    I'm a ${this.species}, ${this.gender}. I have ${this.hands} hands and ${this.legs} legs. My hobby is ${
+      this.hobby
+    }.`;
+  }
+}
+
+class Woman extends Human {
+  constructor(name, age, hobby) {
+    super('female', name, age, hobby);
+    this.greeting = this.greeting.bind(this);
+  }
+  greeting() {
```

Comment:
Your `Woman` and `Man` instance extends from basic `Human` class. Why you declare method `greeting` in two instances ? In OOP common methods must be declared in common class. You can call parent method from `super` instruction.

Reply:
for `Man` and `Woman` instances with method has so many differences, that I decided to divide them.